iroha_p2p/src/peer.rs

```diff
@@ -468,6 +588,21 @@ pub async fn send_message(stream: &mut OwnedWriteHalf, data: &[u8]) -> Result<()
     Ok(())
 }
 
+/// Produces pseudorandom connection id from address
+fn get_connection_id(addr: io::Result<SocketAddr>) -> u64 {
```

Comment:
Wouldn't it lead to id collision and some nasty consequences if several peers have 0 as their id?

Reply:
Generally, they cannot have zeroes, as they have the addresses. If they don't have addresses, then something is wrong with the OS.

Reply:
I see, then in the case of the OS error isn't it better to just drop the connection and log it and maybe reconnect in some time?

Reply:
Have zeros there might lead to debugging these type of errors to be rather painful

Reply:
I think if the OS is so much f*cked up, we'd better to panic and abort all operations :)

Reply:
Ok, then better to panic